langroid/agent/base.py

```diff
@@ -278,17 +278,27 @@ def _get_tool_list(
         if not issubclass(message_class, ToolMessage):
             raise ValueError("message_class must be a subclass of ToolMessage")
         tool = message_class.default_value("request")
+
+        """
+        if tool has handler method explicitly defined - use it,
+        otherwise use the tool name as the handler
+        """
+        if hasattr(message_class, "handler"):
```

Comment:
we should use `_handler` instead of `handler` since we don't want `handler` to be part of the schema passed to the LLM (i.e. we don't want LLM to generate `handler` field). 

Also a note on this should be added to the `chat-agent-tool.md` in the quick-start docs. But let's wait until we know we really need this feature (see comment later)